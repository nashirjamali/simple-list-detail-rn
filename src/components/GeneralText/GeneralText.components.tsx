import React, {FC} from 'react';
import {Text, type TextStyle} from 'react-native';

import type {GeneralTextProps, GeneralTextVariantsTypes} from '../../@types';

import styles from './GeneralText.styles';

/**
 * Retrieves the text style based on the provided variant.
 *
 * @param {GeneralTextVariantsTypes} variant - The variant type for the text style.
 * @returns {TextStyle} The corresponding text style.
 */
const getTextStyle = (variant: GeneralTextVariantsTypes): TextStyle => {
  return styles[variant];
};

/**
 * A general text component that applies styles based on the variant.
 *
 * @param {GeneralTextProps} props - The properties for the GeneralText component.
 * @param {GeneralTextVariantsTypes} props.variant - The variant type for styling.
 * @param {React.ReactNode} props.children - The content to be displayed within the text.
 * @returns {JSX.Element} The rendered text component.
 */
const GeneralText: FC<GeneralTextProps> = ({
  variant,
  children,
  ...props
}: GeneralTextProps): JSX.Element => (
  <Text {...getTextStyle(variant)} {...props}>
    {children}
  </Text>
);

export default GeneralText;
