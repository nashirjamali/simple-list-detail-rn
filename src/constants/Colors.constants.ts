export const COLORS = {
  BRAND: '#FB6544',
  BLACK: '#121212',
  SUCCESS: '#55B585',
  GRAY: '#999999',
  WHITE: '#FFFFFF',
} as const;
