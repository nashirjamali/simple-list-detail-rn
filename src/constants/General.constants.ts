export const API_URL =
  'https://recruitment-test.flip.id/frontend-test' as const;
