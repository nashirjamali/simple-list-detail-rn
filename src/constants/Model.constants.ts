export const TRANSACTION_STATUS = {
  SUCCESS: 'SUCCESS',
  PENDING: 'PENDING',
} as const;
