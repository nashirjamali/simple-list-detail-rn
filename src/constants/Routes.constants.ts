export const ROUTES = {
  TRANSACTION_LIST: 'transactionsList',
  TRANSACTION_DETAILS: 'transactionsDetails',
} as const;
