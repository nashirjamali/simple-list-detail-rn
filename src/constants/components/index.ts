export * from './GeneralText.constants';
export * from './Button.constants';
export * from './Label.constants';
export * from './Card.constants';
