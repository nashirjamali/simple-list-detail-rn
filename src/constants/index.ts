export * from './components';
export * from './Colors.constants';
