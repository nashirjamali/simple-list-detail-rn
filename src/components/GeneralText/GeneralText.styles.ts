import {StyleSheet} from 'react-native';

import {COLORS} from '../../constants';
import type {GeneralTextStyles} from '../../@types';

const {BLACK, BRAND, GRAY} = COLORS;

const styles: GeneralTextStyles = StyleSheet.create({
  button: {
    color: BRAND,
    fontSize: 18,
  },
  header: {
    color: BLACK,
    fontSize: 20,
    fontWeight: 'bold',
  },
  input: {
    color: GRAY,
    fontSize: 16,
  },
  label: {
    fontSize: 16,
  },
  normal: {
    color: BLACK,
    fontSize: 16,
  },
});

export default styles;
