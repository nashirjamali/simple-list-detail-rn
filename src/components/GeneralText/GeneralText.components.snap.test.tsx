import React from 'react';
import {render} from '@testing-library/react-native';

import {GENERAL_TEXT_VARIANTS} from '../../constants';

import GeneralText from './GeneralText.components';

describe('GeneralText Component', () => {
  it('renders correctly with header variant', () => {
    const {toJSON} = render(
      <GeneralText variant={GENERAL_TEXT_VARIANTS.HEADER}>
        This is a header text
      </GeneralText>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with normal variant', () => {
    const {toJSON} = render(
      <GeneralText variant={GENERAL_TEXT_VARIANTS.NORMAL}>
        This is normal text
      </GeneralText>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom style', () => {
    const {toJSON} = render(
      <GeneralText
        variant={GENERAL_TEXT_VARIANTS.NORMAL}
        style={{color: 'red'}}>
        This is normal text with custom style
      </GeneralText>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
