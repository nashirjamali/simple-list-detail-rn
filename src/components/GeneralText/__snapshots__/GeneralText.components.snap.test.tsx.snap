// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralText Component renders correctly with custom style 1`] = `
<Text
  style={
    {
      "color": "red",
      "fontSize": 14,
    }
  }
>
  This is normal text with custom style
</Text>
`;

exports[`GeneralText Component renders correctly with header variant 1`] = `
<Text
  style={
    {
      "color": "#121212",
      "fontSize": 16,
      "fontWeight": "bold",
    }
  }
>
  This is a header text
</Text>
`;

exports[`GeneralText Component renders correctly with normal variant 1`] = `
<Text
  style={
    {
      "color": "#121212",
      "fontSize": 14,
    }
  }
>
  This is normal text
</Text>
`;
